rhizome/host/lib/cert_server_setup.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+require_relative "../../common/lib/util"
+require_relative "../../common/lib/arch"
+require_relative "vm_path"
+require "fileutils"
+require "json"
+
+class CertServerSetup
+  def initialize(vm_name)
+    @vm_name = vm_name
+  end
+
+  def vp
+    @vp ||= VmPath.new(@vm_name)
+  end
+
+  def cert_folder
+    vp.q_cert
+  end
+
+  def cert_path
+    "#{cert_folder}/cert.pem"
+  end
+
+  def key_path
+    "#{cert_folder}/key.pem"
+  end
+
+  def service_name
+    "#{@vm_name}-metadata-endpoint"
+  end
+
+  def service_file_path
+    "/etc/systemd/system/#{service_name}.service"
+  end
+
+  def server_main_path
+    File.join("", "opt", "metadata-endpoint")
+  end
+
+  def vm_server_path
+    File.join(cert_folder, "metadata-endpoint")
+  end
+
+  def package_url
+    if Arch.arm64?
+      "https://github.com/ubicloud/metadata-endpoint/releases/download/v0.1.3/metadata-endpoint_Linux_arm64.tar.gz"
+    elsif Arch.x64?
+      "https://github.com/ubicloud/metadata-endpoint/releases/download/v0.1.3/metadata-endpoint_Linux_x86_64.tar.gz"
+    else
+      fail "BUG: unexpected architecture"
+    end
+  end
+
+  def copy_server
+    unless File.exist?(server_main_path)
+      download_server
+    end
+
+    r "cp #{server_main_path}/metadata-endpoint #{vm_server_path}"
+    r "sudo chown #{@vm_name}:#{@vm_name} #{vm_server_path}"
+  end
+
+  def download_server
+    temp_tarball = "/tmp/metadata-endpoint.tar.gz"
+    r "curl -L3 -o #{temp_tarball} #{package_url}"
+
+    FileUtils.mkdir_p(server_main_path)
+    FileUtils.cd server_main_path do
+      r "tar -xzf #{temp_tarball}"
+    end
+  end
```

Comment:
What happens when two virtual machines download the it at the same time on a fresh VM host?